documentation/design/proposals/authn_authz_for_1.0.adoc

```diff
@@ -0,0 +1,329 @@
+
+==== Messaging Client
+
+When the messaging clients authenticate, in some cases the username must be qualified with an `@DOMAIN` suffix.  The qualification must be used when authenticatiing with an identity from an external LDAP or external AMQP SASL server.
+
+For theinternal identity provider, the domain suffix may be omitted.  The service accounts, the username is always `@@serviceaccount@@`.  For client auth, the identity is the common name from the certificate.
+
+==== Qpid Dispatch Router
+
+This proposal reuses the existing `authServicePlugin` of the Qpid Dispatch Router unchnaged.
+
+=== Resources
+
+The new API will consist of the following types at the infrastructure level:
+
+* `AccessControlService`: used to define an instance of an accesss control service. The `AccessControlService` will define
+the parameters to establish connection with external identity provider (if any).  Selectors will be to configure to which namespaces it should be applied.
+
+* Configuration provided by `MessagingInfra` will define the Access Control Server deployment.
+
+At the tenant level:
+
+* `MessagingTennant`: can be used to specify a specific `AccessControlService` to be used for the tenant.
+
+* `MessagingUser` (v2beta1): used with the internal identity source to provide a messaging username/password pairing.  The password will be stored in the resource a secure fashion.  `MessagingUser` is scoped to the messaging tenant. It cannot
+be used to access any other namespace apart from the only in which it exists.
+
+* `ServiceAccount`: a bearer token of a  `ServiceAccounts` can be used to authenticate to the namespace for messaging.  The service account must exist within the namespace of the messaging tenant.
+
+* `MessagingRole`: define access to one or more addresses.  Its design will be analogous to the kubernetes role object but will not have any relation to kubernetes `roles` or `clusterroles`.
+
+* `MessagingRoleBinding`: grants permission for messaging within the namespace.  A MessagingRoleBinding may reference any `MessagingRole` in the same namespace and any subject produced by the identity providers (user or group).  Its design will be analogous to the kubernetes rolebinding object but will not have any relation to kubernetes `rolebindings` or `clusterrolebindings`.
+
+=== Resources
+
+==== AccessControlService
+
+The `AccessControlService` used to define an instance of an accesss control service.  It allows the configuration the external
+identity providers and gives the ability to disable the internal identity providers.
+
+When the messaging client authenticates the domain part must be provided as part of the SASL username.  An exception
+is made for the client authentication provider and the anonymous provider.
+
+The `AccessControlService` has a selector for namespaces that should use this AccessControlService by default.  This selector
```

Comment:
As I noted earlier, should it be the default, or just available? We need to make sure we have a consistent approach across MessagingPlan, MessagingAddressPlan and AccessControllService for this IMHO.